website/docs/setup/guides/getting-started.md

```diff
@@ -130,62 +203,201 @@ order to print the full event structure:
 <CodeHeader fileName="vector.toml" />
 
 ```toml
+# Consume data
 [sources.foo]
-  type = "stdin"
+  type = "socket"
+  address = "0.0.0.0:9000"
+  mode = "tcp"
 
- # Structure the data
- [transforms.apache_parser]
-   inputs = ["foo"]
-   type = "regex_parser"
-   field = "message"
-   regex = '^(?P<host>[\w\.]+) - (?P<user>[\w]+) (?P<bytes_in>[\d]+) \[(?P<timestamp>.*)\] "(?P<method>[\w]+) (?P<path>.*)" (?P<status>[\d]+) (?P<bytes_out>[\d]+)$'
+# Structure the data
+[transforms.apache_parser]
+  inputs = ["foo"]
+  type = "regex_parser"
+  field = "message"
+  regex = '^(?P<host>[\w\.]+) - (?P<user>[\w]+) (?P<bytes_in>[\d]+) \[(?P<timestamp>.*)\] "(?P<method>[\w]+) (?P<path>.*)" (?P<status>[\d]+) (?P<bytes_out>[\d]+)$'
 
+# Write the data
 [sinks.bar]
-  type = "console"
-    inputs = ["apache_parser"]
-    encoding = "json"
+  inputs = ["apache_parser"]
+  type = "elasticsearch"
+  index = "example-index"
+  host = "http://10.24.32.122:9000"
+
+[[tests]]
+  name = "test apache regex"
+
+  [[tests.inputs]]
+    insert_at = "apache_parser"
+    type = "raw"
+    value = "172.128.80.109 - Bins5273 656 [2019-05-03T13:11:48-04:00] \"PUT /mesh\" 406 10272"
+
+  [[tests.outputs]]
+    extract_from = "apache_parser"
 ```
 </TabItem>
 </Tabs>
 
+This unit test spec has a name, defines an input event to feed into our pipeline
+at a specific transform (in this case our _only_ transform), and defines where
+we expect to see resulting events coming out.
 
-And pipe the same event again through it:
+When we run `vector test ./vector.toml` it will print the input and output
+events of our target transform:
 
-```bash
-echo '172.128.80.109 - Bins5273 656 [2019-05-03T13:11:48-04:00] "PUT /mesh" 406 10272' | vector --config ./vector.toml
+```text
+Running vector.toml tests
+test vector.toml: test apache regex ... passed
+
+inspections:
+
+--- vector.toml ---
+
+test 'test apache regex':
+
+check transform 'apache_parser' payloads (events encoded as JSON):
+  input: {"timestamp":"2020-02-19T14:15:45.997803Z","message":"172.128.80.109 - Bins5273 656 [2019-05-03T13:11:48-04:00] \"PUT /mesh\" 406 10272"}
+  output: {"host":"172.128.80.109","user":"Bins5273","bytes_out":"10272","timestamp":"2019-05-03T13:11:48-04:00","path":"/mesh","bytes_in":"656","method":"PUT","status":"406"}
 ```
 
-Oh snap! This time we get something like:
+This is called an _inspection_ test and is super cool. It allows us to
```

Comment:
Is a flag missing above that would make this an "inspection" test? Or is the vanilla `vector test` command doing this?

Reply:
I see now. Hm. This section feels somewhat long. I'm wondering if we could shorten this step by adding the conditions right off the bat? Maybe add 1 or 2 so it's not overwhelming?

Reply:
I've reworded it to make it clearer. Agree it's long, I'll try and cut it down some more tomorrow. I think it's important to demo the inspections first since that's what we're using instead of manual testing (otherwise we don't see the events).

The alternative is to add a flag `vector test --inspect` that forces all test outputs to be inspections even if they have conditions that pass. Then we can add the full unit test in one step.

Reply:
I'd be in favor of an `--inspect` flag or demonstrating a simple failure. But I would like to shorten this. A getting started guide should be simple and quick to complete.

Reply:
I'll make the regex fail in the guide, that'll allow us to showcase everything with just one test stage.